libs/checkpoint/src/store/postgres.ts

```diff
@@ -0,0 +1,714 @@
+  type GetOperation,
+  type IndexConfig,
+  type SearchItem,
+} from './base.js';
+import { tokenizePath, getTextAtPath } from "./utils.js";
+
+const { Pool } = pkg;
+
+/**
+ * Configuration for vector index in PostgreSQL store.
+ */
+export interface ANNIndexConfig {
+  /**
+   * Type of index to use: 'hnsw' for Hierarchical Navigable Small World, or 'ivfflat' for Inverted File Flat.
+   */
+  kind: "hnsw" | "ivfflat" | "flat";
+
+  /**
+   * Type of vector storage to use.
+   * - 'vector': Regular vectors (default)
+   * - 'halfvec': Half-precision vectors for reduced memory usage
+   */
+  vectorType?: "vector" | "halfvec";
+
+  /**
+   * Maximum number of connections per layer (HNSW only).
+   * @default 16
+   */
+  m?: number;
+
+  /**
+   * Size of dynamic candidate list for index construction (HNSW only).
+   * @default 64
+   */
+  efConstruction?: number;
+
+  /**
+   * Number of inverted lists (clusters) for IVF index (IVFFlat only).
+   */
+  nlist?: number;
+}
+
+/**
+ * Configuration for HNSW (Hierarchical Navigable Small World) index.
+ */
+export interface HNSWConfig extends ANNIndexConfig {
+  kind: "hnsw";
+}
+
+/**
+ * Configuration for IVF Flat index.
+ */
+export interface IVFFlatConfig extends ANNIndexConfig {
+  kind: "ivfflat";
+}
+
+/**
+ * Configuration for vector embeddings in PostgreSQL store with pgvector-specific options.
+ */
+export interface PostgresIndexConfig extends IndexConfig {
+  /**
+   * Specific configuration for the chosen index type (HNSW or IVF Flat)
+   */
+  annIndexConfig?: ANNIndexConfig;
+
+  /**
+   * Distance metric to use for vector similarity search
+   */
+  distanceType?: "l2" | "inner_product" | "cosine";
+}
+
+/**
+ * Row type for database results
+ */
+interface Row {
+  key: string;
+  value: Record<string, unknown>;
+  prefix: string;
+  created_at: Date;
+  updated_at: Date;
+  score?: number;
+}
+
+/**
+ * PostgreSQL-backed store with optional vector search using pgvector.
+ * 
+ * --- Basic setup and usage ---
+ * const store = new PostgresStore({
+ *   connectionString: "postgresql://user:pass@localhost:5432/dbname"
+ * });
+ * await store.setup(); // Run migrations once
+ * 
+ * --- Store and retrieve data ---
+ * await store.put(["users", "123"], "prefs", { theme: "dark" });
+ * const item = await store.get(["users", "123"], "prefs");
+ * 
+ * --- Vector search with embeddings ---
+ * const store = new PostgresStore({
+ *   connectionString: "postgresql://user:pass@localhost:5432/dbname",
+ *   index: {
+ *     dims: 1536,
+ *     embeddings: new OpenAIEmbeddings({ modelName: "text-embedding-3-small" }),
+ *     fields: ["text"] // specify which fields to embed
+ *   }
+ * });
+ * 
+ * --- Store documents ---
+ * await store.put(["docs"], "doc1", { text: "Python tutorial" });
+ * await store.put(["docs"], "doc2", { text: "TypeScript guide" });
+ * 
+ * --- Search by similarity ---
+ * const results = await store.search(["docs"], { query: "programming guides" });
+ */
```

Comment:
Including doc comments like this is great, but please use typedoc comment tags so these code samples render correctly when we generate API docs from them.

https://typedoc.org/documents/Tags._example.html#:~:text=The%20example%20tag%20indicates%20that,how%20to%20use%20the%20function.